src/snowflake/cli/plugins/spcs/image_registry/manager.py

```diff
@@ -66,3 +66,20 @@ def get_registry_url(self):
         if not self._has_url_scheme(sample_repository_url):
             sample_repository_url = f"//{sample_repository_url}"
         return urlparse(sample_repository_url).netloc
+
+    def docker_registry_login(self) -> str:
+        registry_url = self.get_registry_url()
+        token = self.get_token()
+        command = [
+            "docker",
+            "login",
+            "--username",
+            "0sessiontoken",
+            "--password-stdin",
+            registry_url,
+        ]
+        result = subprocess.run(
+            command, input=json.dumps(token), text=True, capture_output=True
+        )
+        result.check_returncode()
```

Comment:
Can we use `check_output`? Also do we need to pass the token via stdin? Can we pass it as "--password" flag?

Reply:
I believe passing token in via stdin is better practice for keeping it out of any shell logs; Docker throws the following warning if you do pass in the password via the --password flag:
<img width="720" alt="image" src="https://github.com/Snowflake-Labs/snowcli/assets/156019931/7be5f863-9c75-48b1-b374-1c1313953895">